pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+
+	var err error
+	var path string
+	var tag string
+
+	cmd := &cobra.Command{
+		Use:   "publish-sdk <package> <language>",
+		Args:  cobra.ExactArgs(2),
+		Short: "Publish a package SDK to supported package registries.",
+		Long:  "Publish a package SDK to supported package registries.",
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validate the language
+
+			lang := args[1]
+
+			switch lang {
+			case "all":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "nodejs":
+				fmt.Println("Nodejs is supported, publishing")
+			case "python":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "java":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "dotnet":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			default:
+				return fmt.Errorf(
+					"language %s is not supported. Must be one of: \n"+
+						"- all \n"+
+						"- nodejs \n"+
+						"- python \n"+
+						"- java \n"+
+						"- dotnet ", lang,
+				)
+			}
+
+			return nil
+		},
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			pkg := args[0]
+			lang := args[1]
+
+			err = publish(pkg, lang, path, tag)
+			if err != nil {
+				return fmt.Errorf("error publishing: %s", err.Error())
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	cmd.PersistentFlags().StringVar(&tag, "tag", "",
+		`The tag for your NPM package, such as "latest""`)
```

Comment:
Is this only for NPM? If so I'd say it should probably be "--npm-tag"

Reply:
Tags are removed per your suggesting below.